### DISCUSSION TITLE
Symfony config builder classes are nice, but recipes are in yaml, ughh

### DISCUSSION THREAD

**Discussion by faizanakram99** [symfony/symfony#42388] [Q&A] [answered]
I recently converted all yaml configs into php and then into new php config builders format.

As an example.

```php
<?php

declare(strict_types=1);

namespace Symfony\Component\DependencyInjection\Loader\Configurator;

use Symfony\Component\HttpFoundation\Session\Storage\Handler\RedisSessionHandler;
use Symfony\Config\FrameworkConfig;

return static function (ContainerConfigurator $containerConfigurator, FrameworkConfig $frameworkConfig): void {
    $frameworkConfig
        ->trustedHosts(['%env(TRUSTED_HOSTS)%'])
        ->secret('%env(APP_SECRET)%')
        ->httpMethodOverride(false)
    ;

    $frameworkConfig
        ->csrfProtection()
        ->enabled(true)
    ;

   // .... more stuff
};
```

The only problem is, everytime a recipe is updated, I need to manually take relevant parts of those yaml files into these php configs.

Shouldn't php be preferred format for recipes now ?
(Now that php is being recommended over yaml)

**Comment by dbrumann** [comment, marked as answer]
As far as I know, this is being worked on. As of now, best practices still suggest the YAML format for application config, which is why it makes sense to deliver the recipes as such.

**Reply by Nyholm** [reply]
Yeah. Any help needed is appreciated. =)

Im considering to create a bundle that can parse the yaml into a php array (this is done by existing logic). Then form that array guess (or reverse engineer) the config builders.

To make all recipes compatible, we also need to consider service definitions. 

Im not sure if this is the "correct" way forward or not.

**Comment by wouterj** [comment]
The config builder is experimental in 5.3 and we never use experimental features in recipes. This will change in 5.4, so maybe recipes will be updated to PHP config builder for 5.4 (or 6.0).

**Reply by faizanakram99** [reply]
ah that makes sense

**Comment by ro0NL** [comment]
does https://github.com/symplify/config-transformer work for you meanwhile?

i can only assume once the format stabilizes, it'll bubble down at the recipe level as well. If that's our prefered format ofc. :) 

IIUC recipe authors are free to choose any format, isnt it? Perhaps in the really long-run above tool can be integrated somewhere, to either input normalize, or output normalize.

**Reply by faizanakram99** [reply]
It only converts to simple php format iirc (`$container->extension('foo', ['foo setting' => ...])`) , not to config builders format (`static fn (FooConfig $fooConfig) => $foo->setting()`).

**Reply by ro0NL** [reply]
if we assume the format stablilizes, rather than being reverted or so. I think one could work on the final conversion step already, or create an issue for it: https://github.com/symplify/symplify/issues

**Reply by faizanakram99** [reply]
Right, sounds like a plan. We will have to wait for 5.4
